assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -125,6 +129,371 @@ export default {
+            }
+          };
+
+          const SidebarChatSection = class extends BaseCustomSidebarSection {
+            @tracked sectionLinks = [];
+
+            constructor() {
+              super(...arguments);
+
+              this.chatService = container.lookup("service:chat");
+              this.chatService.appEvents.on(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.chatService.appEvents.on(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+              this._refreshChannels();
+            }
+
+            teardown() {
+              this.chatService.appEvents.off(
+                "chat:refresh-channels",
+                this._refreshChannels
+              );
+              this.chatService.appEvents.off(
+                "chat:navigated-to-full-page",
+                this._refreshChannels
+              );
+            }
+
+            @bind
+            _refreshChannels() {
+              const newSectionLinks = [];
+              this.chatService.getChannels().then((channels) => {
+                channels.publicChannels.forEach((channel) => {
+                  newSectionLinks.push(
+                    new SidebarChatSectionLink({
+                      channel,
+                      chatService: this.chatService,
+                    })
+                  );
+                });
+                this.sectionLinks = newSectionLinks;
+              });
+            }
+
+            get name() {
+              return "chat-channels";
+            }
+
+            get title() {
+              return I18n.t("chat.chat_channels");
+            }
+
+            get text() {
+              return I18n.t("chat.chat_channels");
+            }
+
+            get actions() {
+              const actions = [
+                {
+                  id: "browseChannels",
+                  title: I18n.t("chat.channels_list_popup.browse"),
+                  action: () => {
+                    this.chatService.router.transitionTo("chat.browse");
+                  },
+                },
+              ];
+              if (this.sidebar.currentUser.staff) {
+                actions.push({
+                  id: "openCreateChannelModal",
+                  title: I18n.t("chat.channels_list_popup.create"),
+                  action: () => {
+                    showModal("create-channel");
+                  },
+                });
+              }
+              return actions;
+            }
+
+            get actionsIcon() {
+              return "cog";
+            }
+
+            get links() {
+              return this.sectionLinks;
+            }
+          };
+
+          return SidebarChatSection;
+        }
+      );
+
+      api.addSidebarSection(
+        (BaseCustomSidebarSection, BaseCustomSidebarSectionLink) => {
+          const SidebarChatSectionLink = class extends BaseCustomSidebarSectionLink {
```

Comment:
I think we can improve the constant name here to reflect the type of section link this is. Otherwise, it is going to be confusing trying to differentiate this sidebar section from the one above when they share the same constant name.